packages/tooltip/src/vaadin-tooltip.js

```diff
@@ -305,14 +355,137 @@ class Tooltip extends ThemePropertyMixin(ElementMixin(PolymerElement)) {
     }
   }
 
-  /** @protected */
-  _open() {
+  /**
+   * Schedule opening the tooltip.
+   * @param {boolean} immediate
+   * @protected
+   */
+  _open(immediate) {
+    if (!immediate && this.__getDelay() > 0 && !this.__closeTimeout) {
+      this.__warmupTooltip();
+    } else {
+      this.__showTooltip();
+    }
+  }
+
+  /**
+   * Schedule closing the tooltip.
+   * @param {boolean} immediate
+   * @protected
+   */
+  _close(immediate) {
+    if (!immediate && this.__getHideDelay() > 0) {
+      this.__scheduleClose();
+    } else {
+      this.__abortClose();
+      this._autoOpened = false;
+    }
+
+    this.__abortWarmUp();
+
+    if (warmedUp) {
+      // Re-start cooldown timer on each tooltip closing.
+      this.__abortCooldown();
+      this.__scheduleCooldown();
+    }
+  }
+
+  /** @private */
+  __getDelay() {
+    return this.delay != null && this.delay > 0 ? this.delay : defaultDelay;
+  }
+
+  /** @private */
+  __getHideDelay() {
+    return this.hideDelay != null && this.hideDelay > 0 ? this.hideDelay : defaultHideDelay;
+  }
+
+  /** @private */
+  __flushClosingTooltips() {
+    closing.forEach((tooltip) => {
+      tooltip._close(true);
+      closing.delete(tooltip);
+    });
+  }
+
+  /** @private */
+  __showTooltip() {
+    this.__abortClose();
+    this.__flushClosingTooltips();
+
     this._autoOpened = true;
+    warmedUp = true;
+
+    // Abort previously scheduled timers.
+    this.__abortWarmUp();
+    this.__abortCooldown();
   }
 
-  /** @protected */
-  _close() {
-    this._autoOpened = false;
+  /** @private */
+  __warmupTooltip() {
+    if (!this._autoOpened) {
+      // First tooltip is opened, warm up.
+      if (!warmUpTimeout && !warmedUp) {
```

Comment:
If we want to support multiple tooltips at the same time, should this `!warmUpTimeout` check also be removed? Or maybe there should be a `const opening = new Set();` also?

Tested by adding global delay and hideDelay in the dev page. With this check in place, the opening delay is bypassed but one of the tooltips doesn't open at all.

Reply:
Removed `!warmUpTimeout` check to re-start warming up while user still moves cursor. 
To me it seems like a better option than keeping references to all the opening tooltips.